src/callbacks_step/save_solution_covariant.jl

```diff
@@ -0,0 +1,106 @@
+# Convert to another set of variables using a solution_variables function that depends on 
+# auxiliary variables
+function convert_variables(u, solution_variables, mesh::P4estMesh{2},
+                           equations, dg, cache)
+    (; aux_node_vars) = cache.auxiliary_variables
+
+    # Extract the number of solution variables to be output 
+    # (may be different than the number of conservative variables) 
+    n_vars = length(solution_variables(Trixi.get_node_vars(u, equations, dg, 1, 1, 1),
+                                       get_node_aux_vars(aux_node_vars, equations, dg, 1, 1,
+                                                         1), equations))
+    # Allocate storage for output
+    data = Array{eltype(u)}(undef, n_vars, nnodes(dg), nnodes(dg), nelements(dg, cache))
+
+    # Loop over all nodes and convert variables, passing in auxiliary variables
+    Trixi.@threaded for element in eachelement(dg, cache)
+        for j in eachnode(dg), i in eachnode(dg)
+            u_node = Trixi.get_node_vars(u, equations, dg, i, j, element)
+            a_node = get_node_aux_vars(aux_node_vars, equations, dg, i, j, element)
+            u_node_converted = solution_variables(u_node, a_node, equations)
```

Comment:
```suggestion
            aux_node = get_node_aux_vars(aux_node_vars, equations, dg, i, j, element)
            u_node_converted = solution_variables(u_node, aux_node, equations)
```
Maybe consequently use `aux` to denote the aux vars, such that users/developers always know that this is an auxiliary data variable?

Reply:
Good idea. This appears in a lot of places, so I have changed it everywhere according to your suggestion.